web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js

```diff
@@ -1,50 +1,95 @@
-//   it under the terms of the GNU Affero General Public License as
-//   published by the Free Software Foundation, either version 3 of the
-//   License, or (at your option) any later version.
-//
-//   This program is distributed in the hope that it will be useful,
-//   but WITHOUT ANY WARRANTY; without even the implied warranty of
-//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-//   GNU Affero General Public License for more details.
-//
-//   You should have received a copy of the GNU Affero General Public License
-//   along with this program.  If not, see <http://www.gnu.org/licenses/>.
-//
-//############################################################################
+/* Copyright 2017 Therp BV, ACSONE SA/NV
+ *  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-openerp.web_ir_actions_act_window_message = function(instance)
+odoo.define('web.web_ir_actions_act_window_message', function(require)
 {
-    instance.web.ActionManager.include({
+    "use strict";
+
+    var ActionManager = require('web.ActionManager'),
+        core = require('web.core'),
+        _ = require('_'),
+        Model = require('web.Model'),
+        Dialog = require('web.Dialog');
+
+    var _t = core._t;
+
+    ActionManager.include({
         ir_actions_act_window_message: function(action, options)
         {
-            var dialog = new instance.web.Dialog(
+            var self = this,
+                buttons = [];
+
+            if(action.close_button_title !== false)
+            {
+                buttons.push({
+                    text: action.close_button_title || _t('Close'),
+                    click: function() {
+                        // refresh the view before closing the dialog
+                        self.inner_widget.active_view
+                            .controller.recursive_reload();
+                        dialog.close()
+                    },
+                    classes: 'btn-default',
+                })
+            }
+
+            var dialog = new Dialog(
                 this,
-                {
-                    size: 'medium',
-                    title: action.title,
-                    buttons: [
-                        {
-                            text: instance.web._t('Close'),
-                            click: function() { dialog.close() },
-                            oe_link_class: 'oe_highlight',
-                        },
-                    ],
-                },
-                jQuery(instance.web.qweb.render(
-                    'web_ir_actions_act_window_message',
+                _.extend(
                     {
-                        'this': this,
-                        'action': action,
-                    }))
+                        size: 'medium',
+                        title: action.title,
+                        $content: $('<div>',
+                            {
+                                text: action.message,
```

Comment:
@zakiuu I think it would be a good idea to wrap line breaks.

```javascript
$content: $('<div>', {
        text: action.message,
        css: {
              'white-space': 'pre-line',
        }
}),
```